src/openrct2/object/Object.h

```diff
@@ -139,11 +139,23 @@ struct ObjectEntryDescriptor
     ObjectGeneration Generation;
     union
     {
+        std::string Identifier; // For JSON objects
```

Comment:
Might just break this union into two fields. The Union is causing a good few issues.